### PR TITLE
[FLINK-22649][python][table-planner-blink] Support StreamExecPythonCalc json serialization/deserialization

### PR DESCRIPTION
**Pull request by HuangXingBo**
## What is the purpose of the change

*This pull request Support StreamExecPythonCalc json serialization/deserialization*

## Brief change log

  - *Support StreamExecPythonCalc json serialization/deserialization*

## Verifying this change

This change added tests and can be verified as follows:

  - *Add json plan test `PythonCalcJsonPlanTest.java`*
  - *Add It test `test_execute_from_json_plan` in `test_udf.py`*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Review by dianfu — APPROVED** [review]

**Comment by dianfu** [comment]
LGTM. 

cc @godfreyhe

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchPhysicalPythonCalc.scala [hunk 1]
```diff
@@ -49,8 +51,10 @@ class BatchPhysicalPythonCalc(
   }
 
   override def translateToExecNode(): ExecNode[_] = {
+    val projection = calcProgram.getProjectList.map(calcProgram.expandLocalRef)
```
Comment (godfreyhe) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchPhysicalPythonCalc.scala:54]: nit: it's better to check the condition in `calcProgram` is empty ?

File: flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/PythonCalcJsonPlanTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        String sinkTableDdl =
+                "CREATE TABLE MySink (\n"
+                        + "  a bigint,\n"
+                        + "  b int\n"
+                        + ") with (\n"
+                        + "  'connector' = 'values',\n"
+                        + "  'table-sink-class' = 'DEFAULT')";
+        tEnv.executeSql(sinkTableDdl);
+        util.verifyJsonPlan("insert into MySink select a, pyFunc(b, b) from MyTable");
+    }
```
Comment (godfreyhe) [inline flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/PythonCalcJsonPlanTest.java:65]: it's better add a test with filter

**Comment by HuangXingBo** [comment]
@godfreyhe Thanks a lot for the review. I have addressed the comments at the latest commit.

**Review by godfreyhe — APPROVED** [review]
thanks for the quick update, LGTM

**Pull request closed without merging**